cmd/lhsm-plugin-az-core/archive.go

```diff
@@ -18,32 +18,56 @@ import (
 )
 
 type ArchiveOptions struct {
-	AccountName   string
-	ContainerName string
-	ResourceSAS   string
-	MountRoot     string
-	BlobName      string
-	SourcePath    string
-	Credential    azblob.Credential
-	Parallelism   uint16
-	BlockSize     int64
-	Pacer         util.Pacer
-	ExportPrefix  string
-	HNSEnabled    bool
-	HTTPClient    *http.Client
+	AccountName     string
+	BlobEndpointURL string
+	DFSEndpointURL  string
+	ContainerName   string
+	ResourceSAS     string
+	MountRoot       string
+	BlobName        string
+	SourcePath      string
+	Credential      azblob.Credential
+	Parallelism     uint16
+	BlockSize       int64
+	Pacer           util.Pacer
+	ExportPrefix    string
+	HNSEnabled      bool
+	HTTPClient      *http.Client
 }
 
-const blobEndPoint string = "https://%s.blob.core.windows.net/"
-const dfsEndPoint string = "https://%s.dfs.core.windows.net/"
 const parallelDirCount = 64 // Number parallel dir metadata uploads
 
 func upload(ctx context.Context, o ArchiveOptions, blobPath string) (_ int64, err error) {
 	p := util.NewPipeline(ctx, o.Credential, o.Pacer, azblob.PipelineOptions{HTTPSender: util.HTTPClientFactory(o.HTTPClient)})
-	cURL, _ := url.Parse(fmt.Sprintf(blobEndPoint+"%s%s", o.AccountName, o.ContainerName, o.ResourceSAS))
-	containerURL := azblob.NewContainerURL(*cURL, p)
-	blobURL := containerURL.NewBlockBlobURL(blobPath)
+	var aclResp *azblob.BlobGetAccessControlResponse
+	sURL, _ := url.Parse(o.BlobEndpointURL + o.ResourceSAS)
+	dfsServiceURL, _ := url.Parse(o.DFSEndpointURL + o.ResourceSAS)
+	blobURL := azblob.NewServiceURL(*sURL, p).NewContainerURL(o.ContainerName).NewBlockBlobURL(blobPath)
+	dfsURL := azblob.NewServiceURL(*dfsServiceURL, p).NewContainerURL(o.ContainerName).NewBlockBlobURL(blobPath)
 	meta := azblob.Metadata{}
 
+	getACLResp := func(path string) (*azblob.BlobGetAccessControlResponse, error) {
+		return dfsURL.GetAccessControl(ctx, nil, nil, nil, nil, nil, nil, nil, nil)
+	}
+
+	setACLResp := func(path string) error {
+		acl := aclResp.XMsACL()
+		owner := aclResp.XMsOwner()
+		group := aclResp.XMsGroup()
+		permissions := aclResp.XMsPermissions()
+		_, err := dfsURL.SetAccessControl(ctx, nil, nil, nil, nil, nil, &acl, nil, nil, nil, nil, nil)
+		if err != nil {
+			util.Log(pipeline.LogError, fmt.Sprintf("Archiving %s. Failed to set Access Control: %s", blobPath, err.Error()))
+			return err
+		}
+		_, err = dfsURL.SetAccessControl(ctx, nil, nil, &owner, &group, &permissions, nil, nil, nil, nil, nil, nil)
```

Comment:
Why 2 set acl calls in a roll? Can't the 4 fields be submitted together?

Reply:
The service does not allow that, this requires two calls

Reply:
Does the second call override the first one's 3 fields?

Reply:
No it doesnt, they are different information.